kml/serdes_gpx.cpp

```diff
@@ -101,20 +103,18 @@ bool GpxParser::Push(std::string_view tag)
   return true;
 }
 
+bool GpxParser::ValidCoordinatesPosition()
+{
+  return GetTagFromEnd(0) == gpx::kWpt || (GetTagFromEnd(0) == gpx::kTrkPt && GetTagFromEnd(1) == gpx::kTrkSeg) ||
+         (GetTagFromEnd(0) == gpx::kRtePt && GetTagFromEnd(1) == gpx::kRte);
```

Comment:
```
auto const lastTag = GetTagFromEnd(0);
if (lastTag == gpx::kWpt)
  return true;
auto const beforeLastTag = GetTagFromEnd(1);
return lastTag == gpx::kTrkPt && beforeLastTag == gpx::kTrkSeg || lastTag == gpx::kRtePt && beforeLastTag == gpx::kRte;
```

Reply:
Fixed

Reply:
This change is not equivalent because we calculate beforeLastTag = GetTagFromEnd(1); for all cases and it fails with assertion when we don't have beforeLastTag tag - this case is now implicitly covered by check that we require GetTagFromEnd(1) only if we are in kTrkPt or kRtePt. Should we leave code as is?

Reply:
You're right! With GetTagFromEnd() signature changed to return `std::string const &`, your code would work optimally.